sys/auto_init/auto_init.c

```diff
@@ -16,653 +16,580 @@
+
+        if (IS_USED(MODULE_CC2420)) {
+            extern void auto_init_cc2420(void);
+            auto_init_cc2420();
+        }
+
+        if (IS_USED(MODULE_ENCX24J600)) {
+            extern void auto_init_encx24j600(void);
+            auto_init_encx24j600();
+        }
+
+        if (IS_USED(MODULE_ENC28J60)) {
+            extern void auto_init_enc28j60(void);
+            auto_init_enc28j60();
+        }
+
+        if (IS_USED(MODULE_ESP_ETH)) {
+            extern void auto_init_esp_eth(void);
+            auto_init_esp_eth();
+        }
+
+        /* don't change the order of auto_init_esp_now and auto_init_esp_wifi */
+        if (IS_USED(MODULE_ESP_NOW)) {
+            extern void auto_init_esp_now(void);
+            auto_init_esp_now();
+        }
+
+        /* don't change the order of auto_init_esp_now and auto_init_esp_wifi */
+        if (IS_USED(MODULE_ESP_WIFI)) {
+            extern void auto_init_esp_wifi(void);
+            auto_init_esp_wifi();
+        }
+
+        if (IS_USED(MODULE_ETHOS)) {
+            extern void auto_init_ethos(void);
+            auto_init_ethos();
+        }
+
+        if (IS_USED(MODULE_DOSE)) {
+            extern void auto_init_dose(void);
+            auto_init_dose();
+        }
+
+        if (IS_USED(MODULE_SLIPDEV)) {
+            extern void auto_init_slipdev(void);
+            auto_init_slipdev();
+        }
+
+        if (IS_USED(MODULE_CC2538_RF)) {
+            extern void auto_init_cc2538_rf(void);
+            auto_init_cc2538_rf();
+        }
+
+        if (IS_USED(MODULE_XBEE)) {
+            extern void auto_init_xbee(void);
+            auto_init_xbee();
+        }
+
+        if (IS_USED(MODULE_KW2XRF)) {
+            extern void auto_init_kw2xrf(void);
+            auto_init_kw2xrf();
+        }
+
+        if (IS_USED(MODULE_USBUS_CDC_ECM)) {
+            extern void auto_init_netdev_cdcecm(void);
+            auto_init_netdev_cdcecm();
+        }
+
+        if (IS_USED(MODULE_NETDEV_TAP)) {
+            extern void auto_init_netdev_tap(void);
+            auto_init_netdev_tap();
+        }
+
+        if (IS_USED(MODULE_SOCKET_ZEP)) {
+            extern void auto_init_socket_zep(void);
+            auto_init_socket_zep();
+        }
+
+        if (IS_USED(MODULE_NORDIC_SOFTDEVICE_BLE)) {
+            extern void gnrc_nordic_ble_6lowpan_init(void);
+            gnrc_nordic_ble_6lowpan_init();
+        }
+
+        if (IS_USED(MODULE_NRFMIN)) {
+            extern void gnrc_nrfmin_init(void);
+            gnrc_nrfmin_init();
+        }
+
+        if (IS_USED(MODULE_W5100)) {
+            extern void auto_init_w5100(void);
+            auto_init_w5100();
+        }
+
+        if (IS_USED(MODULE_SX127X) && !IS_USED(MODULE_SEMTECH_LORAMAC)) {
+            extern void auto_init_sx127x(void);
+            auto_init_sx127x();
+        }
+
+        if (IS_USED(MODULE_NRF802154)) {
+            extern void auto_init_nrf802154(void);
+            auto_init_nrf802154();
+        }
```

Comment:
Can you move those to a separate function while you're at it, so manual init is possible in a generic way?

So you'd have

```C
if (IS_USED(MODULE_AUTO_INIT_GNRC_NETIF)) {
    gnrc_netif_init();
}
```

Where `gnrc_netif_init()` will take care of calling the individual netif init functions.

Reply:
Needs some restructuring of my commit order, but will do.

Reply:
Addressed.